x/gamm/keeper/swap.go

```diff
@@ -123,6 +123,36 @@ func (k Keeper) SwapExactAmountOut(
 	return tokenInAmount, nil
 }
 
+func (k Keeper) CalcOutAmtGivenIn(
```

Comment:
Let's add basic success and failure test cases. To cover given pool not implementing `TraditionalAmmInterface`

Reply:
This is the last point I have not addressed, are you saying instead of returning a blank coin and error, you want me to check if we can type cast the provided poolI to TAMM, and then make a test in swaps_test to test for this?

Reply:
I just want to make sure that we have test coverage for the following:
- balancer pool given -> swap is done (minimal assertions to ensure correctness)
- stableswap pool given -> swap is done (minimal assertions to ensure correctness)
- cl pool given -> swap not done and error returned

Reply:
But this comment is point to Calc, not swap. Was this meant to be for Swap?

Reply:
On a related note, in another PR, I would like to suggest making a helper:

```go
func convertToCFMMPooll() (types.TraditionalAmmInterface, error) {
    	cfmmPool, ok := poolI.(types.TraditionalAmmInterface)
	if !ok {
		return nil, fmt.Errorf("given pool does not implement TraditionalAmmInterface, implements %T", poolI)
	}
       return cfmmPool, nil
}
```

and a similar helper for the concentrated pool conversion

Reply:
@czarcas7ic sorry just saw your reply. I meant calc, not swap. Edited my previous comment